internal/c/libqb/src/mac-mouse-support.cpp

```diff
@@ -0,0 +1,74 @@
+// Mouse support functions for macOS
+// These are required to overcome the limitations of GLUT
+
+#include "libqb-common.h"
+
+#include "mac-mouse-support.h"
+
+#include <ApplicationServices/ApplicationServices.h>
+#include <GLUT/glut.h>
+#include <unistd.h>
+
+#define QB64_EVENT_CLOSE 1
+#define QB64_EVENT_KEY 2
+#define QB64_EVENT_RELATIVE_MOUSE_MOVEMENT 3
+#define QB64_EVENT_FILE_DROP 4
+
+extern "C" int qb64_custom_event(int event, int v1, int v2, int v3, int v4, int v5, int v6, int v7, int v8, void *p1, void *p2);
+void GLUT_MOUSEWHEEL_FUNC(int wheel, int direction, int x, int y);
+
+extern int g_MouseX, g_MouseY;
+
+static CFMachPortRef g_EventTap = nullptr;
+static CFRunLoopSourceRef g_RunLoopSource = nullptr;
+
+static CGEventRef macMouseCallback(CGEventTapProxy proxy, CGEventType type, CGEventRef event, void *userInfo) {
+    auto appPID = reinterpret_cast<int64_t>(userInfo);
+
+    if (CGEventGetIntegerValueField(event, kCGEventTargetUnixProcessID) == appPID) {
+        if (type == kCGEventScrollWheel) {
+            auto deltaScroll = CGEventGetIntegerValueField(event, kCGScrollWheelEventPointDeltaAxis1);
+            GLUT_MOUSEWHEEL_FUNC(0, deltaScroll, g_MouseX, g_MouseY);
+        } else {
+            auto deltaX = CGEventGetIntegerValueField(event, kCGMouseEventDeltaX);
+            auto deltaY = CGEventGetIntegerValueField(event, kCGMouseEventDeltaY);
+            if (deltaX || deltaY)
+                qb64_custom_event(QB64_EVENT_RELATIVE_MOUSE_MOVEMENT, deltaX, deltaY, 0, 0, 0, 0, 0, 0, nullptr, nullptr);
```

Comment:
I think it would be good to pull the `QB64_EVENT_RELATIVE_MOUSE_MOVEMENT` code out of `qb64_custom_event` and place it in a separate function in `libqb.cpp`. After that both `qb64_custom_event()` and your own code can call that new function.

`qb64_custom_event()` was designed so that there's only a single function for the GLUT code to call, but it's pretty nasty to use elsewhere with all the extra parameters and IMO worth avoiding.

Reply:
Cleaned this up as suggested.